src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -82,19 +82,19 @@ private static void LoopReadSocket(int socket)
         {
             while (socket == s_socket)
             {
-                Interop.Sys.NetworkChangeKind kind = Interop.Sys.ReadSingleEvent(socket);
-                if (kind == Interop.Sys.NetworkChangeKind.None)
-                {
-                    continue;
-                }
-                else
+                Interop.Sys.ReadEvents(socket, ProcessEvent);
```

Comment:
This is going to allocate a new NetworkChangeEvent delegate for every call to ReadEvents.  It'd be better to cache that into a static.